rio_tiler/io/stac.py

```diff
@@ -84,8 +73,8 @@ def _get_assets(
         yield asset
 
 
-@dataclass
-class STACReader(BaseReader):
+@dataclass(init=False)
+class STACReader(MultiAssetsReader):
```

Comment:
@geospatial-jeff I'm a bit confused here (I know I wrote the code).
we are defining the abstract class as regular dataclass, but with all variables using `init=False`. 

```python
@dataclass  # type: ignore
class MultiAssetsReader(metaclass=abc.ABCMeta):
    """MultiAssets Reader."""

    bounds: Tuple[float, float, float, float] = field(init=False)
    assets: Sequence[str] = field(init=False)
    reader: Type[BaseReader] = field(init=False)
    reader_options: Optional[Dict] = field(init=False, default_factory=dict)
```
(I'm defining bounds, assets, reader, reader_options here because they will be used in the method, but it's let to the `child` class to fill those)

but when using the class we have to use `@dataclass(init=False)` to no mix the args and kwargs. 

I guess I should just use `@dataclass(init=False)` in the abstract class. 

I'm all 👂 if you have better ideas

Reply:
moved from dataclass to attr which allow this ;-)